### PR TITLE
[Feature] Light visibility

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [x] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

**Comment by troublemaker52025** [comment]
rebuild

**Review by jk20012001 — APPROVED** [review]

File: cocos/3d/lights/directional-light-component.ts [hunk 1]
```diff
@@ -111,6 +111,19 @@ export class DirectionalLight extends Light {
         }
     }
 
+    /**
+     * @en Visibility mask of the light, declaring a set of node layers that will be visible to this light(directional has not).
+     * @zh 光照的可见性掩码，声明在当前光照中可见的节点层级集合（该功能不包括方向光的设置）。
```
Comment (minggo) [inline cocos/3d/lights/directional-light-component.ts]: 该功能不包括方向光的设置 -> 对方向光不起效

还得补充一下关闭方向光

File: cocos/3d/lights/light-component.ts [hunk 2]
```diff
@@ -233,6 +237,21 @@ export class Light extends Component {
         }
     }
 
+    /**
+     * @en Visibility mask of the light, declaring a set of node layers that will be visible to this light.
+     * @zh 光照的可见性掩码，声明在当前光照中可见的节点层级集合。
```
Comment (minggo) [inline cocos/3d/lights/light-component.ts]: Does directional light be effected? If not, should mention it too.

File: cocos/3d/lights/directional-light-component.ts (hunk 1 above)
Comment (minggo) [reply cocos/3d/lights/directional-light-component.ts]: Light also has this property, why need to override it?

File: cocos/3d/lights/light-component.ts (hunk 2 above)
Comment (troublemaker52025) [reply cocos/3d/lights/light-component.ts]: sure

File: native/tools/tojs/scene.ini [hunk 3]
```diff
@@ -99,7 +99,7 @@ getter_setter = Root::[device:_device mainWindow curWindow tempWindow windows sc
        PassInstance::[parent],
        Node::[uuid angle matrix hasChangedFlags/getChangedFlags/setChangedFlags _persistNode/isPersistNode/setPersistNode mobility/getMobility/setMobility],
        Ambient::[skyColor skyIllum groundAlbedo enabled/isEnabled mipmapCount],
-       Light::[baked/isBaked/setBaked color useColorTemperature/isUseColorTemperature/setUseColorTemperature colorTemperature colorTemperatureRGB node type name scene],
+       Light::[baked/isBaked/setBaked color useColorTemperature/isUseColorTemperature/setUseColorTemperature colorTemperature colorTemperatureRGB visibility node type name scene],
```
Comment (minggo) [inline native/tools/tojs/scene.ini]: Please don't use .ini, use .i instead. Since v3.7, engine uses swig instead.

Comment (troublemaker52025) [reply native/tools/tojs/scene.ini]: sure

File: cocos/3d/lights/directional-light-component.ts (hunk 1 above)
Comment (troublemaker52025) [reply cocos/3d/lights/directional-light-component.ts]: 图片

We have discussed

**Review by minggo — APPROVED** [review]

**Pull request merged**